lib/amqp/gen/consumer.ex

```diff
@@ -178,14 +180,28 @@ defmodule Amqpx.Gen.Consumer do
 
   def terminate(_, %__MODULE__{channel: nil}), do: nil
 
-  def terminate(_, %__MODULE__{channel: channel}) do
-    if Process.alive?(channel.pid) do
-      Channel.close(channel)
+  def terminate(reason, %__MODULE__{channel: channel}) do
+    case reason do
+      :normal -> close_channel(channel)
+      :shutdown -> close_channel(channel)
+      {:shutdown, _} -> close_channel(channel)
+      _ -> :ok
```

Comment:
shouldn't we close the channel regardless of the reason? :thinking:

Reply:
Actually, the channel is always closed because it is linked. However, for "normal" closures, we need to add some logic, since as you can see, amqp_client does not consider `{:shutdown, term}` or simply `:shutdown` as normal closures